### PR TITLE
mgr/cephadm: When create osd which include db, _create_daemon will be…

### PR DESCRIPTION
**Pull request by strenuous-life**
… executed twice. So we should filter the db type

Signed-off-by: jianglong01 <jianglong01@qianxin.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by strenuous-life** [comment]
jenkins test make check

File: src/pybind/mgr/cephadm/tests/test_cephadm.py [hunk 1]
```diff
@@ -1239,7 +1239,8 @@ def test_osd_activate(self, _run_cephadm, cephadm_module: CephadmOrchestrator):
                     'tags': {
                         'ceph.cluster_fsid': cephadm_module._cluster_fsid,
                         'ceph.osd_fsid': 'uuid'
-                    }
+                    },
+                    'type': 'data'
                 }]
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/tests/test_cephadm.py:1244]: Great, you needed to modify the tests already. Can you also add a test that contains `'type': 'db'`?

**Comment by sebastian-philipp** [comment]
please apply:

```diff
diff --git a/src/pybind/mgr/cephadm/tests/test_cephadm.py b/src/pybind/mgr/cephadm/tests/test_cephadm.py
index 72e365b345..37cdf3d470 100644
--- a/src/pybind/mgr/cephadm/tests/test_cephadm.py
+++ b/src/pybind/mgr/cephadm/tests/test_cephadm.py
@@ -1277,5 +1277,14 @@ Traceback (most recent call last):
                 }]
             }
             _run_cephadm.return_value = (json.dumps(ceph_volume_lvm_list), '', 0)
+            _run_cephadm.reset_mock()
             assert cephadm_module._osd_activate(
                 ['test']).stdout == "Created osd(s) 1 on host 'test'"
+            assert _run_cephadm.mock_calls == [
+                mock.call('test', 'osd', 'ceph-volume',
+                          ['--', 'lvm', 'list', '--format', 'json'], no_fsid=False, image=''),
+                mock.call('test', 'osd.1', 'deploy',
+                          ['--name', 'osd.1', '--meta-json', mock.ANY,
+                              '--config-json', '-', '--osd-fsid', 'uuid'],
+                          stdin=mock.ANY, image=''),
+            ]
```
to verify this actually works. otherwise this now looks good to me.

**Comment by strenuous-life** [comment]
jenkins retest this please

**Review by sebastian-philipp — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-06-13_08:30:20-rados-wip-kefu-testing-2021-06-13-1215-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-06-13_13:23:30-rados-wip-kefu-testing-2021-06-13-2001-distro-basic-smithi/

**Pull request merged**